spec/features/friendship_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'rails_helper'
+
+RSpec.describe 'Friendship features', type: :request do
+  scenario 'Create and accept new friendship' do
+    visit new_user_registration_path
+    fill_in 'user_name', with: 'tadeu'
+    fill_in 'user_email', with: 'tadeu@tadeu.com'
+    fill_in 'user_password', with: '123123'
+    fill_in 'user_password_confirmation', with: '123123'
+    click_button 'Sign up'
+    expect(page).to have_content('Welcome! You have signed up successfully.')
+
+    click_link 'Sign out'
+    expect(page).to have_content('You need to sign in or sign up before continuing.')
+
+    visit new_user_registration_path
+    fill_in 'user_name', with: 'maria'
+    fill_in 'user_email', with: 'maria@maria.com'
+    fill_in 'user_password', with: '123123'
+    fill_in 'user_password_confirmation', with: '123123'
+    click_button 'Sign up'
+    expect(page).to have_content('Welcome! You have signed up successfully.')
+
+    visit users_path
+    click_link 'Add friend'
+    expect(page).to have_content('Friendship request sent!')
```

Comment:
Instead of doing this every time to create new users, you could use a method such as the one used in [this project](https://github.com/voscarmv/ror-social-scaffold/blob/master/spec/features/friendship_spec.rb), where you only have to create the users directly in the database once before each test. This will keep your code DRY and make your tests more maintainable, as you won't have to change all the places in the code every time you want to make some change.

You should use something like this to create the users only once for this type of test:

```
require 'rails_helper'

RSpec.describe 'Creating friendship invitations', type: :feature do
  before :each do
    user1 = User.create(name: 'Oscar', email: 'o@mail.com', password: '123456')
    user2 = User.create(name: 'Alexis', email: 'a@mail.com', password: '123456')
    user3 = User.create(name: 'Mememan', email: 'm@mail.com', password: '123456')
    user4 = User.create(name: 'Ivan', email: 'i@mail.com', password: '123456')
    user1.posts.create(content: 'A new post by Oscar.')
    user3.posts.create(content: 'A new post by Mememan.')
    user4.posts.create(content: 'A new post by Ivan.')
    f1 = Friendship.create(requesting_friend: user4, requested_friend: user2, status: false) # request sent
    f1.update(status: true) # request accepted
    f2 = Friendship.create(requesting_friend: user2, requested_friend: user3, status: false) # request sent
    f2.update(status: true) # request accepted

    visit user_session_path
    fill_in 'Email', with: 'a@mail.com'
    fill_in 'Password', with: '123456'
    click_on 'Log in'
  end
...
```

In order to make good use of this style of testing, you will also probably need a [database cleaner method.](https://github.com/voscarmv/ror-social-scaffold/blob/master/spec/support/database_cleaner.rb) Add it as well if necessary.